src/providers/FusionAuthProvider.tsx

```diff
@@ -22,29 +25,46 @@ export const FusionAuthProvider: React.FC<Props> = ({
     clientID,
     scope,
     redirectUri,
+    idTokenHint,
     children,
 }) => {
     const login = useCallback(
         async (state = '') => {
-            console.log('TEST 1');
-            const fullUrl = await generateUrl(
+            const queryParams = {
+                client_id: clientID,
+                scope: scope,
+                response_type: 'code',
+                redirect_uri: redirectUri,
+                code_challenge: await generatePKCE(),
+                code_challenge_method: 'S256',
+                state: `${generateRandomString()}:${state}`,
+            };
+            const fullUrl = generateUrl(
                 FunctionType.login,
                 baseUrl,
-                clientID,
-                scope,
-                redirectUri,
-                state,
+                queryParams,
             );
             window.location.assign(fullUrl);
         },
         [baseUrl, clientID, scope, redirectUri],
     );
 
+    const logout = useCallback(async () => {
+        const queryParams = {
+            client_id: clientID,
+            post_logout_redirect_uri: redirectUri,
+            id_token_hint: idTokenHint || '',
```

Comment:
It's more semantic to use null coalescing operator here:
```suggestion
            id_token_hint: idTokenHint ?? '',
```